daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/ScenarioRunner.scala

```diff
@@ -421,21 +426,30 @@ object ScenarioRunner {
         case SResultError(err) =>
           SubmissionError(Error.RunnerException(err), enrich(onLedger.incompleteTransaction))
         case SResultNeedContract(coid, committers, callback) =>
-          ledger.lookupContract(coid, committers, readAs, callback) match {
-            case Left(err) => SubmissionError(err, enrich(onLedger.incompleteTransaction))
-            case Right(_) => go()
+          discTable.contractById.get(coid) match {
```

Comment:
do you want to add a todo somewhere to resolve disclosed contracts in speedy at a later point to avoid the duplication between Daml Engine & this?

Reply:
I expect the duplication to go away, when we move the disclosed contracts logic deeper into the speedy machine. But I'll add an issue.